source/Calamari.Tests/Fixtures/PackageRetention/LeastFrequentlyUsedWithAgingCacheAlgorithmFixture.cs

```diff
@@ -0,0 +1,160 @@
+                                       {
+                                           CreateEntry("package-1", "1.0", 10, ("task-1", 1)),
+                                           CreateEntry("package-1", "1.1", 10, ("task-2", 1)),
+                                           CreateEntry("package-2", "1.0", 10, ("task-3", 1))
+                                       },
+                                       10,
+                                       new ExpectedPackageIdAndVersion("package-1", "1.0"));
+
+            yield return SetUpTestCase("WhenOnlyRelyingOnAge_ThenReturnOldestThatTakesEnoughSpace",
+                                       new[]
+                                       {
+                                           CreateEntry("package-1", "1.0", 10, ("task-1", 1)),
+                                           CreateEntry("package-2", "1.0", 10, ("task-2", 10))
+                                       },
+                                       10,
+                                       new ExpectedPackageIdAndVersion("package-1"));
+
+            yield return SetUpTestCase("WhenOnlyRelyingOnHitCount_ThenReturnPackageWithFewestHits",
+                                       new[]
+                                       {
+                                           CreateEntry("package-1", "1.0", 10, ("task-1", 1), ("task-3", 1)),
+                                           CreateEntry("package-2", "1.0", 10, ("task-2", 1))
+                                       },
+                                       10,
+                                       new ExpectedPackageIdAndVersion("package-2"));
+
+            yield return SetUpTestCase("WhenOnlyRelyingOnNewerVersions_ThenReturnPackagesWithMostNewVersions",
+                                       new[]
+                                       {
+                                           CreateEntry("package-1", "1.0", 10),
+                                           CreateEntry("package-1", "1.1", 10),
+                                           CreateEntry("package-1", "1.2", 10),
+                                           CreateEntry("package-1", "1.3", 10),
+                                           CreateEntry("package-2", "1.0", 10),
+                                           CreateEntry("package-2", "1.1", 10),
+                                           CreateEntry("package-2", "1.2", 10)
+                                       },
+                                       30,
+                                       new ExpectedPackageIdAndVersion("package-1", "1.0"),
+                                       new ExpectedPackageIdAndVersion("package-1", "1.1"),
+                                       new ExpectedPackageIdAndVersion("package-2", "1.0"));
+        }
+
+        static TestCaseData SetUpTestCase(string testName, JournalEntry[] testJournalEntries, int spaceNeeded, params ExpectedPackageIdAndVersion[] expectedPackageIdAndVersions)
```

Comment:
Created this for readability - When using the TestCaseData constructor, there's no easy way to see which parameter is which.  This just gives them names.